### ISSUE TITLE
Keyboard navigation is not selecting the dropdown options

### ISSUE THREAD

**Issue by IsaSih** [rancher/dashboard#13262]
**Setup**
- Rancher version: 
Rancher | v2.11-a1bda936c809e0394520c9d40d2998529223401e-head
Dashboard | master 7119928
Rancher CLI | v2.11.0-rc.1

**Describe the bug**
When using the keyboard navigation on the charts catalog page, the tab key skips the options inside the dropdown lists. Hence, the user cannot filter extensions by repository. Also, the keyboard navigation stops working.

**To Reproduce**
1- From the dashboard UI, select a cluster
2- Then, go to Apps > Charts
3- Use the keyboard navigation to access the options inside the dropdown lists
4- Try to proceed with the keyboard navigation

**Result**
The tab key skips the options inside the dropdown lists. Also, the keyboard navigation stops working.

**Expected Result**
Keyboard navigation must be able to select the options inside the dropdown lists and user must be able to continue the navigation to other elements of the UI

**Screenshots**

https://github.com/user-attachments/assets/1860eb71-6810-4acc-90b9-9f21777d49e0

**Comment by aalves08** [comment]
@IsaSih can you explain a bit more about the problem you encountered? Is it related to tabbing into the dropdown and it doesn't open automatically? 🤔

**Comment by IsaSih** [comment]
Tabbing into the dropdown displays the dropdown options, but tabbing stops working after this. Not only the user cannot tab into the options, but also not anywhere else in the page. The keyboard navigation stops working completely when the charts dropdown is expanded.

**Comment by aalves08** [comment]
According to a conversation on Slack, this should already be fixed. Just needs double-checking, hence I moved it to test and assigned it to @izaac and @IsaSih  🙏 

FYI @rak-phillip

**Comment by izaac** [comment]
@aalves08 @rak-phillip what are the expected keys in this component to check and uncheck the select options? I see this works with `Enter` and it's usually how a select dropdown usually works. But this custom element's options are checkboxes and checkboxes select/unselect use the `Space` key.

**Comment by aalves08** [comment]
@izaac We've addressed this as a [combobox](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/). If you look at the chapter about `Listbox Popup Keyboard Interaction`, we don't have any action for the `space` key, although the usecase there is slightly different, but in essence the component is perceived and "coded" as a combobox.

**Comment by izaac** [comment]
@aalves08 I see, yes it's a quite customized List thus I was looking for the actual expected behavior. 
This is what I was looking for, thanks.

**Comment by izaac** [comment]
Rancher version: `v2.11-13170b249a26f304b1ae89a96688f547a96b77b1-head`
UI version: `master 28b6f80`

Reopening as there are javascript errors on certain actions ❌ 

Reproducible in Chrome and Brave. When in the list I deselect the last selected list item a console error is thrown on console and can't uncheck it.

The issue is in the second half of the recording, but I've kept the first half as it's not always reproducible once the page is already loaded an interacted with.

 
https://github.com/user-attachments/assets/fcf171b7-7ed7-4c44-98f9-7b0a3385052c

**Comment by aalves08** [comment]
@izaac I merged a fix for https://github.com/rancher/dashboard/issues/13262#issuecomment-2683180400. It's ready for another round of testing 🙏

**Comment by izaac** [comment]
Rancher version: `v2.11-41098eee7331faf5d4a478dfcf66232867ed6a6f-head`
UI Version: `master 95c0062`
Firefox version: `135.0.1`
Chrome version: `133.0.6943.141`

Validated working on this Rancher/UI version. ✅
